src/metabase/api/database.clj

```diff
@@ -264,9 +266,13 @@
   []
   (saved-cards-virtual-db-metadata :include-tables? true, :include-fields? true))
 
-(defn- db-metadata [id]
+(defn- db-metadata [id include-hidden-tables]
   (-> (api/read-check Database id)
       (hydrate [:tables [:fields [:target :has_field_values] :has_field_values] :segments :metrics])
+      (update :tables (fn [tables]
+                        (if include-hidden-tables
+                          tables
+                          (filter #(nil? (:visibility_type %)) tables))))
```

Comment:
You could do 

```clj
(update :tables (if include-hidden-tables?
                  identity
                  (partial remove :visibility_type)))
```

here to make it a bit more concise. `(remove pred coll)` is the same as `(filter (complement pred) coll)`.